serde/src/private/de.rs

```diff
@@ -1262,6 +1262,17 @@ mod content {
         {
             match self.content {
                 Content::Unit => visitor.visit_unit(),
+
+                // As a special case, allow deserializing newtype variant containing unit. E.G:
+                // #[derive(Deserialize)]
+                // #[serde(tag = "result")]
+                // enum Response<T> {
+                //     Success(T),
+                // }
+                //
+                // We want {"result": "Success"} to deserialize into `Response<T>`.
+                Content::Map(ref v) if v.is_empty() => visitor.visit_unit(),
+                Content::Seq(ref v) if v.is_empty() => visitor.visit_unit(),
```

Comment:
I think I see the `Content::Map` case corresponds to the `serialize_map` that has been added to serde/src/private/ser.rs in this PR. But what is `Content::Seq` handling?

Reply:
When I first read your comment, I thought that The line with `Content::Seq` should not be there and I had made a mistake. I read the code for unit structs at the function below and somehow just thought that I should copy both lines, with `Content::Map` and `Content::Seq`. However, that would mean that my example enum above could be deserialized from `["success"]`, which may not be what we want. And I made a commit to fix this.

But after a bit more investigation, I see that the behaviour of unit structs seem to be just that. I.E. if we have a unit struct `SomeStruct` we could deserialize `Response::Success(SomeStruct)` from `["success"]`. So for consistency this should perhaps also be allowed for units so that `()` and unit structs behave in the same way.

What do you think? Should we allow the sequence notation or not? If we decide to allow it, I will just revert my last commit.

Reply:
I think we should land it like this and find out whether anyone needs the `Seq` case in practice.

Thanks!